src/freenas/usr/local/lib/middlewared_truenas/plugins/failover_/iptables_linux.py

```diff
@@ -0,0 +1,157 @@
+from middlewared.service import Service, accepts, job, CallError
+
+import subprocess
+import copy
+
+
+V4_FILE = '/data/v4-fw.rules'
+V6_FILE = '/data/v6-fw.rules'
+JOB_LOCK = 'firewall_rules_update'
+
+
+class IptablesService(Service):
+
+    class Config:
+        namespace = 'failover.firewall'
+        private = True
+
+    async def generate_default_rules(self, data):
+        """
+        Generate a list of default firewall rules.
+        """
+
+        # this is always the first rule
+        rules = ['*filter']
+
+        # the positions of these are important
+        rules.insert(1, ':INPUT ACCEPT [0:0]')
```

Comment:
Why do we need to use `insert` here and not just declare it at `L24` ?

Reply:
I used insert to be explicit about their positions in the list.

Reply:
We can just add a comment for that perhaps ? In my opinion, it just looks better that way as there is no dynamic change happening here otherwise.

Reply:
Also iirc `ACCEPT` is a terminating target, meaning that we having this at the start, will have all the packets not move forward to the other rules. Please correct me otherwise ?

Reply:
`ACCEPT` is the default behavior for the `INPUT` chain. It means if no rules match in the `INPUT` chain, allow the traffic.

It's the path of least resistance since changing the default behavior for the `INPUT` chain to `DROP` would be system wide and would effect everything running service on the system.

Reply:
I agree on that, i am concerned about the order. Having that rule as the first one means that every packet is going to be accepted and rest of the rules in the chain are not going to be evaluated. If i am correct, it means we should have this at the end so that other rules are processed first.

Reply:
No that line in the list is for the default `INPUT` chain not a rule. Rules only get generated if `drop_all` is run.

So the rules look like this:
`-P INPUT ACCEPT
-P FORWARD ACCEPT
-P OUTPUT ACCEPT
-A INPUT -p tcp -m tcp --dport 22 -j ACCEPT
-A INPUT -p tcp -m tcp --dport 80 -j ACCEPT
-A INPUT -p tcp -m tcp --dport 443 -j ACCEPT
-A INPUT -s 192.168.1.184/32 -j DROP
-A INPUT -s 2.2.2.2/32 -j DROP
-A INPUT -s 10.10.10.10/32 -j DROP`

These rules mean allow any traffic to dst ports 22, 80, and 443 no matter what. If the traffic doesn't match those rules, then go to next rule which means drop any traffic to the IP of `192.168.1.184` (which is a VIP address on HA systems)

Reply:
Finally if the traffic doesn't match any of the `-A INPUT` rules, then allow the traffic since `-P INPUT ACCEPT`